src/server/scripts/Shadowlands/SanctumOfDomination/boss_sylvanas_windrunner.cpp

```diff
@@ -0,0 +1,5615 @@
+                });
+
+                scheduler.Schedule(75s + 172ms, [this](TaskContext /*task*/)
+                {
+                    me->SendPlaySpellVisualKit(SPELL_VISUAL_KIT_SYLVANAS_INTRODUCTION_TALK_10, 0, 0);
+                });
+
+                scheduler.Schedule(78s + 422ms, [this](TaskContext /*task*/)
+                {
+                    DoCastSelf(SPELL_RANGER_BOW_STANCE, false);
+                });
+
+                scheduler.Schedule(78s + 437ms, [this](TaskContext /*task*/)
+                {
+                    me->SetHomePosition(me->GetPosition());
+
+                    instance->SetData(DATA_SYLVANAS_INTRO, DONE);
+                });
+
+                break;
+            }
+
+            case ACTION_WINDRUNNER_MODEL_ACTIVATE:
+            {
+                me->m_Events.AddEvent(new PauseAttackState(me, true), me->m_Events.CalculateTime(1));
+                DoCastSelf(SPELL_WINDRUNNER_DISAPPEAR_01, false);
+                break;
+            }
+
+            case ACTION_WINDRUNNER_MODEL_DEACTIVATE:
+            {
+                me->m_Events.AddEvent(new PauseAttackState(me, false), me->m_Events.CalculateTime(1));
+                me->SetNameplateAttachToGUID(ObjectGuid::Empty);
+                me->RemoveAura(SPELL_WINDRUNNER_DISAPPEAR_01);
+                break;
+            }
+
+            case ACTION_RESET_MELEE_KIT:
+            {
+                _meleeKitCombo = 0;
+                break;
+            }
+
+            case ACTION_RANGER_SHOT:
+            {
+                DoCastVictim(SPELL_RANGER_SHOT, false);
+
+                if (!_rangerShotOnCD)
+                    _rangerShotOnCD = true;
+
+                _meleeKitCombo = 0;
+
+                scheduler.Schedule(750ms, [this](TaskContext /*task*/)
+                {
+                    me->resetAttackTimer();
+                });
+
+                scheduler.Schedule(1s + 750ms, [this](TaskContext /*task*/)
+                {
+                    if (!me->IsWithinMeleeRange(me->GetVictim()))
+                        DoAction(ACTION_RANGER_SHOT);
+                    else
+                        _rangerShotOnCD = false;
+                });
+                break;
+            }
+
+            case ACTION_PREPARE_INTERMISSION:
+            {
+                _specialEvents.CancelEventGroup(1);
+                events.CancelEventGroup(1);
+
+                me->m_Events.AddEvent(new PauseAttackState(me, true), me->m_Events.CalculateTime(1));
+
+                _specialEvents.SetPhase(PHASE_INTERMISSION);
+                events.SetPhase(PHASE_INTERMISSION);
+                events.ScheduleEvent(EVENT_DOMINATION_CHAINS, 1s, PHASE_INTERMISSION);
+                break;
+            }
+
+            case ACTION_HAUNTING_WAVE_SECOND_CHAIN:
+            {
+                if (_hauntingWaveTimes < 5)
+                {
+                    scheduler.Schedule(250ms, [this](TaskContext /*task*/)
```

Comment:
this whole action has way too much duplicate code in it, merge it and differentiate between _hauntingWaveTimes < 5

Reply:
time to give it a look; let's see how I can diminish the rows.